miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/ProjectPage.java

```diff
@@ -2,36 +2,118 @@
+    public static final String SHORTNAME = "shortName";
+    public static final String DESCRIPTION = "description";
+    public static final String PROGRESS = "progress";
+    public static final String REFERENCE_GENOME = "referenceGenome";
+
+    private Fields() {
+      throw new IllegalStateException("Util class not intended for instantiation");
+    }
+  }
+
   @FindBy(id = "projectId")
   private WebElement idLabel;
-
   @FindBy(id = "name")
   private WebElement nameLabel;
+  @FindBy(id = "creationDate")
+  private WebElement creationDateLabel;
+  @FindBy(id = "alias")
+  private WebElement aliasLabel;
+  @FindBy(id = "shortName")
+  private WebElement shortNameLabel;
+  @FindBy(id = "description")
+  private WebElement descriptionLabel;
+  @FindBy(name = "progress")
+  private List<WebElement> progressLabel;
+  @FindBy(id = "referenceGenome")
+  private WebElement referenceGenomeLabel;
+  @FindBy(id = "save")
+  private WebElement saveButton;
 
   public ProjectPage(WebDriver driver) {
     super(driver);
     PageFactory.initElements(driver, this);
     waitWithTimeout().until(titleContains("Project "));
   }
 
-  public static ProjectPage get(WebDriver driver, String baseUrl, long projectId) {
-    driver.get(baseUrl + "miso/project/" + projectId);
+  public static ProjectPage get(WebDriver driver, String baseUrl, Long projectId) {
+    driver.get(baseUrl + "miso/project/" + (projectId == null ? "new" : projectId));
     return new ProjectPage(driver);
   }
 
-  public String getProjectId() {
+  public String getId() {
     return idLabel.getText();
   }
 
-  public String getProjectName() {
+  public String getName() {
     return nameLabel.getText();
   }
 
+  public String getCreationDate() {
+    return creationDateLabel.getText();
+  }
+
+  public void setCreationDate(String date) {
+    setText(date, creationDateLabel);
+  }
+
+  public String getAlias() {
+    return aliasLabel.getAttribute("value");
+  }
+
+  public void setAlias(String alias) {
+    setText(alias, aliasLabel);
+  }
+
+  public String getShortName() {
+    return shortNameLabel.getAttribute("value");
+  }
+
+  public void setShortName(String shortName) {
+    setText(shortName, shortNameLabel);
+  }
+
+  public String getDescription() {
+    return descriptionLabel.getAttribute("value");
+  }
+
+  public void setDescription(String description) {
+    setText(description, descriptionLabel);
+  }
+
+  public String getProgress() {
+    return getSelectedRadioButtonText(progressLabel);
+  }
+
+  public void setProgress(String progress) {
+    setRadioButton(progress, progressLabel);
+  }
+
+  public String getReferenceGenome() {
+    return getSelectedDropdownText(referenceGenomeLabel);
+  }
+
+  public void setReferenceGenome(String referenceGenome) {
+    setDropdown(referenceGenome, referenceGenomeLabel);
+  }
+
+  public void clickSave() {
+    saveButton.click();
+    waitWithTimeout().until(titleContains("Project "));
```

Comment:
This won't actually do anything since the title still matches before the page is reloaded. It seems that since the page can take a couple seconds to actually start reloading, the current `waitForPageRefresh` method is similarly doing nothing though - `document.readyState` is `complete` already, and we don't know that it's ever been in any other state.

Should probably at least change this to `waitForPageRefresh()` though. Then it can be fixed in one place at least.

This method (and any that cause page reload) should return a new Page object since the page may have changed significantly or failed to load, and there may be stale elements, new elements, etc.